### PR TITLE
[onert] Support regular transpose with Transpose op

### PR DESCRIPTION
**Pull request by ragmani**
For #4053

This commit makes Transpose operation supports regular transpose.
  - Support regular transpose
  - Add nnfw tests for Transpose op
  - Enable some nnapi tests for regular transpose

Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/one_op_tests/Transpose.cc [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  _context->setBackends({"acl_cl", "acl_neon", "cpu"});
+  _context->setCompileFail();
+
+  SUCCEED();
+}
+
+TEST_F(GenModelTest, neg_OneOp_Transpose_InvalidPermsVal)
+{
+  CircleGen cgen;
+  std::vector<int32_t> perms_data{-3, 3, 1, 2};
```
Comment (hyunsik-yoon) [inline tests/nnfw_api/src/one_op_tests/Transpose.cc:97]: :+1:

File: runtime/onert/backend/cpu/ops/TransposeLayer.cc [hunk 2]
```diff
@@ -38,10 +39,22 @@ template <typename T> void TransposeLayer::transpose()
 {
   nnfw::cker::TransposeParams param;
   assert(_perm->num_dimensions() == 1);
-  param.perm_count = _perm->dimension(0);
-  for (auto i = 0; i < param.perm_count; i++)
+
+  param.perm_count = _input->num_dimensions();
+  if (_perm->dimension(0) == 0) // Regular Transpose
```
Comment (hyunsik-yoon) [inline runtime/onert/backend/cpu/ops/TransposeLayer.cc:44]: When I first see this PR, I was curious about the definition of "regular transpose".
I googled the definition but it was hard to find. Could you explain a bit?

File: runtime/onert/core/src/compiler/OperationValidator.cc [hunk 3]
```diff
@@ -238,7 +238,8 @@ void OperationValidator::visit(const ir::operation::Transpose &node)
   const auto &output_shape = _ctx.at(output_index).shape();
   const auto &input_shape = _ctx.at(input_index).shape();
 
-  OP_REQUIRES(input_shape.rank() == static_cast<int>(_ctx.at(perm_index).shape().num_elements()));
+  OP_REQUIRES((_ctx.at(perm_index).shape().num_elements() == 0 && input_shape.rank() == 2) ||
```
Comment (hyunsik-yoon) [inline runtime/onert/core/src/compiler/OperationValidator.cc:241]: I wonder whether `&& input_shape.rank() == 2` is necessary.

File: runtime/onert/backend/cpu/ops/TransposeLayer.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/cpu/ops/TransposeLayer.cc:44]: It is not a officially term.
In linear algebra, the transpose of a matrix is an operator which flips a matrix over its diagonal. It switches the row and column indices of the matrix A by producing another matrix.
In neural networks, it is expanded in multiple dimensions with a vector specifying how the multiple dimensions will be permuted. If the vector is regularly set to (n-1...0), it is the same as expanded transpose of linear algebra and as Transpose operation of nnapi without perms. So I commented it as "Regular Transpose".

Comment (hyunsik-yoon) [reply runtime/onert/backend/cpu/ops/TransposeLayer.cc:44]: I see. 
I am afraid that the term "Regular Transpose" may _not_ be meaningful to most people.
Why don't you remove `Regular Transpose` throughout the code?

Comment (ragmani) [reply runtime/onert/backend/cpu/ops/TransposeLayer.cc:44]: Okay I will change it to (n-1...0)

File: tests/nnfw_api/src/one_op_tests/Transpose.cc [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      {{0.f,  1.f,   2.f,   3.f,   4.f,   20.f,  21.f, 22.f,  23.f,  24.f,  40.f,  41.f,
+        42.f, 43.f,  44.f,  60.f,  61.f,  62.f,  63.f, 64.f,  80.f,  81.f,  82.f,  83.f,
+        84.f, 100.f, 101.f, 102.f, 103.f, 104.f, 5.f,  6.f,   7.f,   8.f,   9.f,   25,
+        26,   27,    28,    29,    45.f,  46.f,  47.f, 48.f,  49.f,  65.f,  66.f,  67.f,
+        68.f, 69.f,  85.f,  86.f,  87.f,  88.f,  89.f, 105.f, 106.f, 107.f, 108.f, 109.f,
+        10.f, 11.f,  12.f,  13.f,  14.f,  30.f,  31.f, 32.f,  33.f,  34.f,  50.f,  51.f,
+        52.f, 53.f,  54.f,  70.f,  71.f,  72.f,  73.f, 74.f,  90.f,  91.f,  92.f,  93.f,
+        94.f, 110.f, 111.f, 112.f, 113.f, 114.f, 15.f, 16.f,  17.f,  18.f,  19.f,  35.f,
+        36.f, 37.f,  38.f,  39.f,  55.f,  56.f,  57.f, 58.f,  59.f,  75.f,  76.f,  77.f,
+        78.f, 79.f,  95.f,  96.f,  97.f,  98.f,  99.f, 115.f, 116.f, 117.f, 118.f, 119.f}}));
```
Comment (wateret) [inline tests/nnfw_api/src/one_op_tests/Transpose.cc:53]: FYI, you can leave `.f`s out 😃

Comment (ragmani) [reply tests/nnfw_api/src/one_op_tests/Transpose.cc:53]: Okay I will remove it.

**Comment by ragmani** [comment]
@nnfw-bot test android-runtime-builder

**Comment by ragmani** [comment]
@nnfw-bot test android-runtime-builder

**Review by wateret — APPROVED** [review]
LGTM

**Review by hyunsik-yoon — APPROVED** [review]
LGTM

**Pull request merged**